src/greeter/GreeterApp.cpp

```diff
@@ -59,6 +60,16 @@ namespace SDDM {
 
     GreeterApp *GreeterApp::self = nullptr;
 
+    // catch key-press or mouse-click to restart the timer that resets the form
+    bool GreeterApp::notify(QObject *receiver, QEvent *event) {
+        if (event->type() == QEvent::KeyPress || event->type() == QEvent::MouseButtonPress) {
+            m_formResetTimer.stop();
```

Comment:
just m_formResetTimer.start()
see docs of QTimer

but more importantly, there's a better way to do event filtering.

Currently you're watching *every* single QObject in your entire application. When you move the mouse, you're restarting the timer a bunch of times.

What you want is to run installEventFilter(this) when we create the QQuickViews.
Then change this method to override eventFilter instead of notify